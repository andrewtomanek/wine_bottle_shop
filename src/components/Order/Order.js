import React from "react";

import classes from "./Order.module.css";

const Order = (props) => {
  const inventory = [];
  for (let inventoryName in props.inventory) {
    inventory.push({
      name: props.listItems[inventoryName],
      amount: props.inventory[inventoryName],
    });
  }

  const inventoryOutput = inventory.map((ig) => {
    return (
      <div className={classes.OrderItem} key={ig.name}>
        <p className={classes.OrderText}>{ig.name}</p>
        <p className={classes.OrderAmount}>{ig.amount}</p>
      </div>
    );
  });

  return (
    <div className={classes.OrderBox}>
      <div className={classes.OrderInventory}>{inventoryOutput}</div>
      <p className={classes.OrderPrice}>
        Cena {Number.parseFloat(props.price).toFixed(2)}
      </p>
    </div>
  );
};

export default Order;
