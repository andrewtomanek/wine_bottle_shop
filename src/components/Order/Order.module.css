.Order {
  display: grid;
  grid-auto-flow: row;
  align-content: space-around;
  justify-content: space-around;
  padding: 0.3rem 1rem;
}

.OrderBox {
  display: grid;
  grid-auto-flow: row;
  align-content: space-around;
  justify-content: space-around;
  padding: 0.3rem 1rem;
  border: 0.2rem solid var(--red);
  box-shadow: 2px 2px 2px #966909;
}

.OrderInventory {
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-template-rows: repeat(2, 1fr);
  align-content: space-around;
  justify-content: space-around;
  padding: 0.3rem 1rem;
}

.OrderItem {
  outline: none;
  color: var(--red);
  font-family: inherit;
  text-align: center;
  font-size: 1.2rem;
  font-weight: 600;
  padding: 0.1rem 0.5rem;
  margin: 0rem;
}

.OrderText {
  background-color: var(--wine);
  color: var(--red);
  font-size: 1rem;
  font-weight: 700;
  padding: 0.1rem 0.2rem;
}

.OrderPrice {
  background-color: var(--wine);
  border: 0.2rem solid var(--red);
  color: var(--red);
  font-size: 1rem;
  font-weight: 700;
  padding: 0.1rem 0.2rem;
  box-shadow: 2px 2px 2px #966909;
}
