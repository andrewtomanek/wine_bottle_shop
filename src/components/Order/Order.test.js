import React from "react";
import { shallow } from "enzyme";
import Order from "./Order";

const purchaseCancelled = jest.fn();
const purchaseContinued = jest.fn();

const props = {
  inventory: {
    shopItem1: 1,
    shopItem2: 2,
    shopItem3: 3,
    shopItem4: 4,
    shopItem5: 5,
    shopItem6: 6,
    shopItem7: 7,
    shopItem8: 8,
  },
  listItems: {
    shopItem1: "Solaris",
    shopItem2: "Dornfelder",
    shopItem3: "Agni",
    shopItem4: "Muscaris",
    shopItem5: "Sauvignon",
    shopItem6: "Tramín",
    shopItem7: "Hibernal",
    shopItem8: "Pálava",
  },
  price: 999,
  purchaseCancelled,
  purchaseContinued,
};

describe("Shallow Render Order component", () => {
  it("renders Sum", () => {
    const wrapper = shallow(<Order {...props} />);
    expect(wrapper.find("p").find(".OrderPrice").text()).toContain(
      "Cena 999.00"
    );
  });
  it("renders order items", () => {
    const wrapper = shallow(<Order {...props} />);
    expect(wrapper.find(".OrderInventory").children()).toHaveLength(8);
  });

  it("Match Snapshot of Order component", () => {
    const wrapper = shallow(<Order {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
