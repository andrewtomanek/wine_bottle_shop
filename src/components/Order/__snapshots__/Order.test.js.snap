// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Render Order component Match Snapshot of Order component 1`] = `
<div
  className="OrderBox"
>
  <div
    className="OrderInventory"
  >
    <div
      className="OrderItem"
      key="Solaris"
    >
      <p
        className="OrderText"
      >
        Solaris
      </p>
      <p
        className="OrderAmount"
      >
        1
      </p>
    </div>
    <div
      className="OrderItem"
      key="Dornfelder"
    >
      <p
        className="OrderText"
      >
        Dornfelder
      </p>
      <p
        className="OrderAmount"
      >
        2
      </p>
    </div>
    <div
      className="OrderItem"
      key="Agni"
    >
      <p
        className="OrderText"
      >
        Agni
      </p>
      <p
        className="OrderAmount"
      >
        3
      </p>
    </div>
    <div
      className="OrderItem"
      key="Muscaris"
    >
      <p
        className="OrderText"
      >
        Muscaris
      </p>
      <p
        className="OrderAmount"
      >
        4
      </p>
    </div>
    <div
      className="OrderItem"
      key="Sauvignon"
    >
      <p
        className="OrderText"
      >
        Sauvignon
      </p>
      <p
        className="OrderAmount"
      >
        5
      </p>
    </div>
    <div
      className="OrderItem"
      key="Tramín"
    >
      <p
        className="OrderText"
      >
        Tramín
      </p>
      <p
        className="OrderAmount"
      >
        6
      </p>
    </div>
    <div
      className="OrderItem"
      key="Hibernal"
    >
      <p
        className="OrderText"
      >
        Hibernal
      </p>
      <p
        className="OrderAmount"
      >
        7
      </p>
    </div>
    <div
      className="OrderItem"
      key="Pálava"
    >
      <p
        className="OrderText"
      >
        Pálava
      </p>
      <p
        className="OrderAmount"
      >
        8
      </p>
    </div>
  </div>
  <p
    className="OrderPrice"
  >
    Cena 
    999.00
  </p>
</div>
`;
